tests/components/qwikswitch/test_init.py

```diff
@@ -106,3 +120,299 @@ async def test_sensor_device(hass, aioclient_mock):
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    await hass.async_start()
+    await hass.async_block_till_done()
+    await asyncio.sleep(0.01)
+    assert hass.states.get("sensor.ss1")
+    assert not hass.states.get("sensor.ss2")
+    assert hass.states.get("sensor.ss3")
+    listen_mock.stop()
+
+
+async def test_non_binary_sensor_with_binary_args(
+    hass, aioclient_mock, qs_devices, caplog
+):
+    """Test that the system logs a warning when a non-binary device has binary specific args."""
+
+    config = {
+        "qwikswitch": {
+            "sensors": [
+                {
+                    "name": "ss1",
+                    "id": "@a00001",
+                    "channel": 1,
+                    "type": "qwikcord",
+                    "invert": True,
+                },
+            ]
+        }
+    }
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=qs_devices)
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    await hass.async_start()
+    await hass.async_block_till_done()
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    assert hass.states.get("sensor.ss1")
+    assert "invert should only be used for binary_sensors" in caplog.text
+    listen_mock.stop()
+
+
+async def test_non_relay_switch(hass, aioclient_mock, qs_devices, caplog):
+    """Test that the system logs a warning when a switch is configured for a device that is not a relay."""
+
+    config = {"qwikswitch": {"switches": ["@a00003"]}}
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=qs_devices)
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    await hass.async_start()
+    await hass.async_block_till_done()
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    assert not hass.states.get("switch.dim_3")
+    assert "You specified a switch that is not a relay @a00003" in caplog.text
+    listen_mock.stop()
+
+
+async def test_unknown_device(hass, aioclient_mock, qs_devices, caplog):
+    """Test that the system logs a warning when a network device has unknown type."""
+
+    config = {"qwikswitch": {}}
+    qs_devices[1]["type"] = "ERROR_TYPE"
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=qs_devices)
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    await hass.async_start()
+    await hass.async_block_till_done()
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    assert hass.states.get("light.switch_1")
+    assert not hass.states.get("light.light_2")
+    assert hass.states.get("light.dim_3")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5403678)